code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -182,66 +182,69 @@
+				. = FALSE
+			else if(atmos_requirements["max_oxy"] && oxy > atmos_requirements["max_oxy"])
+				. = FALSE
+			else if(atmos_requirements["min_tox"] && tox < atmos_requirements["min_tox"])
+				. = FALSE
+			else if(atmos_requirements["max_tox"] && tox > atmos_requirements["max_tox"])
+				. = FALSE
+			else if(atmos_requirements["min_n2"] && n2 < atmos_requirements["min_n2"])
+				. = FALSE
+			else if(atmos_requirements["max_n2"] && n2 > atmos_requirements["max_n2"])
+				. = FALSE
+			else if(atmos_requirements["min_co2"] && co2 < atmos_requirements["min_co2"])
+				. = FALSE
+			else if(atmos_requirements["max_co2"] && co2 > atmos_requirements["max_co2"])
+				. = FALSE
+		else
+			if(atmos_requirements["min_oxy"] || atmos_requirements["min_tox"] || atmos_requirements["min_n2"] || atmos_requirements["min_co2"])
+				. = FALSE
+
+	if(check_temp)
+		var/areatemp = get_temperature(environment)
+		if((areatemp < minbodytemp) || (areatemp > maxbodytemp))
+			. = FALSE
 
+
+/mob/living/simple_animal/handle_environment(datum/gas_mixture/environment)
 	var/atom/A = src.loc
 	if(isturf(A))
-		var/turf/T = A
 		var/areatemp = get_temperature(environment)
 		if( abs(areatemp - bodytemperature) > 40 )
 			var/diff = areatemp - bodytemperature
 			diff = diff / 5
 			//world << "changed from [bodytemperature] by [diff] to [bodytemperature + diff]"
 			bodytemperature += diff
 
-		if(isopenturf(T))
-			var/turf/open/ST = T
-			if(ST.air)
-				var/ST_gases = ST.air.gases
-				ST.air.assert_gases(arglist(hardcoded_gases))
-
-				var/tox = ST_gases["plasma"][MOLES]
-				var/oxy = ST_gases["o2"][MOLES]
-				var/n2  = ST_gases["n2"][MOLES]
-				var/co2 = ST_gases["co2"][MOLES]
-
-				ST.air.garbage_collect()
-
-				if(atmos_requirements["min_oxy"] && oxy < atmos_requirements["min_oxy"])
-					atmos_suitable = 0
-				else if(atmos_requirements["max_oxy"] && oxy > atmos_requirements["max_oxy"])
-					atmos_suitable = 0
-				else if(atmos_requirements["min_tox"] && tox < atmos_requirements["min_tox"])
-					atmos_suitable = 0
-				else if(atmos_requirements["max_tox"] && tox > atmos_requirements["max_tox"])
-					atmos_suitable = 0
-				else if(atmos_requirements["min_n2"] && n2 < atmos_requirements["min_n2"])
-					atmos_suitable = 0
-				else if(atmos_requirements["max_n2"] && n2 > atmos_requirements["max_n2"])
-					atmos_suitable = 0
-				else if(atmos_requirements["min_co2"] && co2 < atmos_requirements["min_co2"])
-					atmos_suitable = 0
-				else if(atmos_requirements["max_co2"] && co2 > atmos_requirements["max_co2"])
-					atmos_suitable = 0
-
-				if(!atmos_suitable)
-					adjustBruteLoss(unsuitable_atmos_damage)
-
-		else
-			if(atmos_requirements["min_oxy"] || atmos_requirements["min_tox"] || atmos_requirements["min_n2"] || atmos_requirements["min_co2"])
-				adjustBruteLoss(unsuitable_atmos_damage)
+	if(!environment_is_safe(environment))
+		adjustBruteLoss(unsuitable_atmos_damage)
```

Comment:
this should probably be adjustHealth so it doesn't pass through brute damage multipliers